addons/website/controllers/main.py

```diff
@@ -234,6 +241,37 @@ def customize_template_get(self, key, full=False, bundles=False):
         return request.registry["ir.ui.view"].customize_template_get(
             request.cr, request.uid, key, full=full, bundles=bundles,
             context=request.context)
+
+    @http.route('/website/save_css', type='json', auth='user', website=True)
+    def save_css(self, css):
+        ir_att = request.registry['ir.attachment']
+        file_name = css['id'][7:] if css['id'].startswith('/custom') else css['id']
+        css_file = ir_att.search(request.cr, request.uid, [('url', "=", '/custom' + file_name)])
+        ir_att.invalidate_bundle(request.cr, request.uid, type='css', context=request.context)
+        if css_file:
+            ir_att.write(request.cr, request.uid, css_file, {'datas': css['text'].encode('utf-8').encode('base64')}, context=request.context)
+        else:
+            ir_att.create(request.cr, request.uid, dict(
+                datas=css['text'].encode('utf-8').encode('base64'),
+                mimetype='text/css',
+                type='binary',
+                name=file_name,
+                datas_fname=css['id'].split('/')[-1],
+                url='/custom'+css['id'],
+            ), context=request.context)
+        return True
+
+    @http.route('/website/remove_css', type='json', auth='user', website=True)
+    def remove_css(self, file_path):
+        file = request.env['ir.attachment'].search([('url', "=", file_path)])
+        if file:
+            file.sudo().unlink()
```

Comment:
Please don't apply sudo without check. It's unsecured.
Any user can use the controller and delete all attachments.